sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeExtractors.scala

```diff
@@ -210,4 +203,45 @@ case class GpuArrayContains(left: Expression, right: Expression)
     lhs.getBase.listContainsColumn(rhs.getBase)
 
   override def prettyName: String = "array_contains"
-}
\ No newline at end of file
+}
+
+object GetArrayItemUtil {
+  def evalColumnar(array: GpuColumnVector, ordinal: Scalar, dataType: DataType,
+                   zeroIndexed: Boolean): ColumnVector = {
```

Comment:
I asked for a common implementation because I thought it could be combined cleanly.  This is not combined cleanly. At a minimum we need some documentation to explain exactly what is happening because it is not obvious from just the API. However, I am inclined to admin that my request was a bad one because the corner cases are handled differently enough that it is not simple to combine them, and perhaps we should leave these separate.